install/aws-installation-guide.md

```diff
@@ -0,0 +1,143 @@
+# Install Kata Containers on Amazon Web Services 
+
+* [Install and Configure AWS CLI](#install-and-configure-aws-cli)
+* [Create an EC2 SSH key pair](#create-an-ec2-ssh-key-pair)
+* [Launch i3.metal instance](#launch-i3metal-instance)
+* [Install Kata](#install-kata)
+
+Kata Containers on Amazon Web Services (AWS) makes use of [i3.metal](https://aws.amazon.com/ec2/instance-types/i3/) instances. Most of the installation procedure is identical to that for Kata on your preferred distribution, except that you have to run it on bare metal instances since AWS doesn't support nested virtualization yet. This guide walks you through creating an i3.metal instance.
+
+## Install and Configure AWS CLI
+
+### Requirements
+
+* Python:
+  * Python 2 version 2.6.5+
+  * Python 3 version 3.3+
+
+### Install
+
+Install with this command:
+
+```bash
+$ pip install awscli --upgrade --user
+```
+
+### Configure
+
+First, verify it:
+
+```bash
+$ aws --version
+```
+
+Then configure it:
+
+```bash
+$ aws configure
+```
+
+Specify the required parameters:
+
+```
+AWS Access Key ID []: <your-key-id-from-iam>
+AWS Secret Access Key []: <your-secret-access-key-from-iam>
+Default region name []: <your-aws-region-for-your-i3-metal-instance>
+Default output format [None]: <yaml-or-json-or-empty>
+```
+
+Alternatively, you can create the files: `~/.aws/credentials` and `~/.aws/config`:
+
+```bash
+$ cat <<EOF > ~/.aws/credentials
+[default]
+aws_access_key_id = <your-key-id-from-iam>
+aws_secret_access_key = <your-secret-access-key-from-iam>
+EOF
+$ cat <<EOF > ~/.aws/config
+[default]
+region = <your-aws-region-for-your-i3-metal-instance>
+EOF
+```
+
+For more information on how to get AWS credentials please refer to [this guide](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_access-keys.html). Alternatively, you can ask the administrator of your AWS account to issue one with the AWS CLI:
+
+```bash
+aws iam create-access-key --user-name=<your-aws-username>
+```
+
+More general AWS CLI guidelines can be found [here](https://docs.aws.amazon.com/cli/latest/userguide/installing.html).
+
+## Create an EC2 SSH key pair
+
+You will need this to access your instance.
+
+```bash
+$ aws ec2 create-key-pair --key-name MyKeyPair \
+                          | grep KeyMaterial \
```

Comment:
`jq` is nice but IMO, `cut` and `tr` are mode widely available than `jq`. Does anybody have any other opinions?

Reply:
I agree that `cut` and `tr` are pretty much guaranteed to be available, but I do also agree that `jq` would make the logic simpler and the more robust so I'm going to +1 the idea of using [`jq`](https://github.com/stedolan/jq/) [1], particularly since it's available in the standards repos for all the common distros.

--- 

[1] OOI does anyone know anything about why it's been over *three years* since the last `jq` github release? Since this is clearly still an active project, this strikes me as somewhat odd.

Reply:
For me it depends on how complex the JSON coming back from the command is - if it returns a pretty simple one line or a simple layout that is guaranteed to only have one line that you will match with `KeyMaterial`, then I've no issue with the `cut` and `tr`. But, just for such reasons, I have used `jq` elsewhere because the data coming back was more complex, and I needed stronger guarantees of parsing out the exact item I was looking for :-)

Nope, no idea why `jq` has not had a formal release for so long :-)

Reply:
AFAIK does not come with most of the distros base installation, so you'll have to at least mention that as a prerequisite.

Reply:
`jq` is pretty stable, so I guess they haven't felt like a new release, but it's that their last release was in 2015.

@raravena80 `jq` -1 (for this case `cut` and `tr` seem simpler)
@jodh-intel `jq` +1
@grahamwhaley `jq` -1?
@marcov `jq`? -1?
@nmeyerhans `jq` +1